### PR TITLE
Dynamic resource count per UUID entry based on file size and configurable bucket size

### PR DESCRIPTION
**Pull request by TomWindels**
With these changes, it should be possible to group multiple resources into a single UUID, based on a target file size per resource. The bucket size is now more dynamic as well.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,6 @@
 node_modules/
 data/
+package.json
```
Comment (woutslabbinck) [inline .gitignore:3]:
```suggestion

```
package.json is quite important to install the packages to run the project. I would not add this to .gitignore

File: EventSource/index.ts [hunk 2]
```diff
@@ -19,85 +19,101 @@
...
-            // add platform to resource
-            const platform = store.getQuads(sens.object, 'http://www.w3.org/ns/sosa/isHostedBy', null, null);
-            platform.forEach((plat) => {
-                quads = quads.concat(
-                    store.getQuads(plat.object, null, null, null)
-                );
-            });
-        });
-
-        resources.push(quads)
```
Comment (woutslabbinck) [inline EventSource/index.ts:96]: I see this part is something you do not need from the `index.ts` script. However, for the original pipeline it is necessary for the extraction of the full resource of the location model.

To be more concrete:
Without that bit of code I would per resource receive this:
```ttl
<http://location.example.com/tracks/observation/2022-08-07T08%3A14%3A04Z> dct:isVersionOf ex:location ;
rdf:type sosa:Observation ;
sosa:hasFeatureOfInterest <https://data.knows.idlab.ugent.be/person/woslabbi/#me> ;
sosa:hasResult <http://location.example.com/tracks/observation/result/2022-08-07T08%3A14%3A04Z> ;
sosa:hasSimpleResult "POINT(3.621189000 50.962510000)"^^geo:wktLiteral ;
sosa:madeBySensor <http://sensor.be> ;
sosa:observedProperty <http://location.example.com/location> ;
sosa:resultTime "2022-08-07T08:14:04Z"^^xsd:dateTime .

<http://location.example.com/tracks/observation/result/2022-08-07T08%3A14%3A04Z> rdf:type sosa:Result ;
wgs:elevation "7.1" ;
wgs:latitude "50.962510000" ;
wgs:longitude "3.621189000" ;
<https://w3id.org/transportmode#transportMode> <https://w3id.org/transportmode#Walking> .
```
While with this piece of code I receive more information:
```ttl
<http://device.be> rdf:type sosa:Platform ;
sosa:hosts <http://sensor.be> .

<http://location.example.com/location> rdf:type sosa:observedProperty ;
rdfs:comment "The Geographic location observed by a sensor."@en ;
rdfs:label "Location"@en .

<http://location.example.com/tracks/observation/2022-08-07T08%3A14%3A04Z> dct:isVersionOf ex:location ;
rdf:type sosa:Observation ;
sosa:hasFeatureOfInterest <https://data.knows.idlab.ugent.be/person/woslabbi/#me> ;
sosa:hasResult <http://location.example.com/tracks/observation/result/2022-08-07T08%3A14%3A04Z> ;
sosa:hasSimpleResult "POINT(3.621189000 50.962510000)"^^geo:wktLiteral ;
sosa:madeBySensor <http://sensor.be> ;
sosa:observedProperty <http://location.example.com/location> ;
sosa:resultTime "2022-08-07T08:14:04Z"^^xsd:dateTime .

<http://location.example.com/tracks/observation/result/2022-08-07T08%3A14%3A04Z> rdf:type sosa:Result ;
wgs:elevation "7.1" ;
wgs:latitude "50.962510000" ;
wgs:longitude "3.621189000" ;
<https://w3id.org/transportmode#transportMode> <https://w3id.org/transportmode#Walking> .

<http://sensor.be> rdf:type sosa:Sensor ;
sosa:isHostedBy <http://device.be> ;
sosa:madeObservation <http://location.example.com/tracks/observation/2022-08-07T08%3A14%3A04Z> ;
sosa:observes <http://location.example.com/location> .

<https://data.knows.idlab.ugent.be/person/woslabbi/#me> rdf:type sosa:FeatureOfInterest .
```
As a suggestion, the above code could be placed in a utility function `extractLocationResource`.
The default behaviour would still be to call that resource.
In your case, you are only interested in samples per subject, so then you can extract the resource on subject base (which can also be configurable).

**Review by woutslabbinck — CHANGES_REQUESTED** [review]
Overall seems good. I have on issue about the extraction of resources, to which I have added a suggestion such that the original pipeline would still produce the same result.

If you have another suggestion, feel free to add :)
There are many options for us both to get what we want.

File: .gitignore (hunk 1 above)
Comment (TomWindels) [reply .gitignore:3]: Ah okay. I'm not very experienced in using Node.js and Typescript, so I was unaware of this. It suddenly got added on my end as an untracked file, so I thought that it wasn't supposed to be present in the repo.

Comment (woutslabbinck) [reply .gitignore:3]: Yeah, I know which one you are talking about. 
This one gets installed due to the install.sh script. 
Maybe we should just add that generated package.json to github.

File: EventSource/index.ts (hunk 2 above)
Comment (TomWindels) [reply EventSource/index.ts:96]: I find it strange that the recursive implementation didn't add the triples with <http://sensor.be> (and recursively <http://device.be> and <http://location.example.com/location>) as a subject. I'll play around with it some more to find out how this happened and see if I can resolve it to do this properly as well. However, the triple with subject <https://data.knows.idlab.ugent.be/person/woslabbi/#me> would indeed not be added with this approach, so a separate function (that is called on default, when no additional arguments are present) would be required indeed.

Comment (TomWindels) [reply EventSource/index.ts:96]: I have created a fix for another (related) issue, but I am unable to replicate your specific results. Using the data you have given as an example above, and creating a TTL file from that, the data gets parsed to a single resource, which contains the following subjects (and all its data):
Set(6) {
  'http://location.example.com/tracks/observation/2022-08-07T08%3A14%3A04Z',
  'http://location.example.com/location',
  'https://data.knows.idlab.ugent.be/person/woslabbi/#me',
  'http://location.example.com/tracks/observation/result/2022-08-07T08%3A14%3A04Z',
  'http://sensor.be',
  'http://device.be'
}`. Could you maybe provide an example .ttl (or .nt) file so I can see if that helps in replicating the issue?

**Review by woutslabbinck — APPROVED** [review]
Nice Work

**Pull request merged**